src/cryptoadvance/specter/util/common.py

```diff
@@ -29,6 +29,53 @@ def snake_case2camelcase(word):
     return "".join(x.capitalize() or "_" for x in word.split("_"))
 
 
+def replace_substring(text, start_position, replace_length, new_str):
+    return text[:start_position] + new_str + text[start_position + replace_length :]
+
+
+def satamount_formatted(value):
```

Comment:
Wouldn't it make sense to rename this to sth. more generic, like `convert_to_sats`or simply `satamount` and add a check whether the value is a float or a string and then return `"{:,.0f}".format(round(float(value) * 1e8))` (if value is a string) or  `"{:,.0f}".format(round(value * 1e8))` (if value is a float). We could then also reuse this function in other filters like `btcunitamount`.

Reply:
> return `"{:,.0f}".format(round(float(value) * 1e8))` (if value is a string) or `"{:,.0f}".format(round(value * 1e8))` (if value is a float).

Isn't this already covered with the current implementation?

> Wouldn't it make sense to rename this to sth. more generic, like `convert_to_sats`or simply `satamount`

OK.

Reply:
I wouldn't take chances with floats, they are scary, perhaps you are right, I would rather make it more explicit. 
If you want to stress that the function returns a string, rather use type hints. I can change it if you want and push it. I am also writing tests as I go through the review.

Reply:
> I can change it if you want and push it.

great.  I think later we can add also coloring.

Reply:
Shouldn't use floats at all. The `Decimal` class and its `quantize` method are the only precise way to handle digits below the decimal point.

Reply:
> Wouldn't it make sense to rename this to sth. more generic, like `convert_to_sats`or simply `satamount` and add a check whether the value is a float or a string and then return `"{:,.0f}".format(round(float(value) * 1e8))` (if value is a string) or `"{:,.0f}".format(round(value * 1e8))` (if value is a float). We could then also reuse this function in other filters like `btcunitamount`.

Either case sounds like a mess. I'm not familiar with how the `value` data is passed around by the time it gets here, but values should internally be represented as integer sats. If some other call further back is changing ints to floats or strings, then instead of having two interpretations (int -> float -> str; int -> str -> str), the prior formatting functions should be altered or avoided so that this is the only conversion to str.

Reply:
The original source of the amounts is Bitcoin Core which uses floats.